src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -564,6 +564,15 @@ QgsOgrProvider::QgsOgrProvider( QString const &uri, const ProviderOptions &optio
     CSLDestroy( papszTokens );
   }
 
+  // Older versions of GDAL incorrectly report that shapefiles support
+  // DateTime.
+#if GDAL_VERSION_NUM <= GDAL_COMPUTE_VERSION(3,1,2)
```

Comment:
The GDAL fix will appear in 3.2. I'm a bit hesitant to backport it in the 3.1 series, so :
```suggestion
#if GDAL_VERSION_NUM < GDAL_COMPUTE_VERSION(3,2,0)
```